bitcoin/src/blockdata/transaction.rs

```diff
@@ -698,15 +698,20 @@ impl Transaction {
                 .collect(),
             output: self.output.clone(),
         };
-        cloned_tx.txid().into()
+        cloned_tx.compute_txid().into()
     }
 
+    #[deprecated(since = "0.31.0", note = "use compute_txid() instead")]
```

Comment:
Is it a good form to explain "previous name hid the fact that this operation is CPU-expensive"? Might be less annoying if the users that need to fix it know the rationale.

Reply:
The commit message explains the rational: https://github.com/rust-bitcoin/rust-bitcoin/pull/2366/commits/b62d25eca60568287f04310bb8867ed8957f5a78.  Maybe that's enough?

Reply:
@yancyribbens it's annoying to have to dig into git blame of a different repository for that. And I think many people would be annoyed by not seeing the point of rename, so strongly agree that putting the reason there would be great.

Reply:
Yeap. You update dependency, run `cargo build` and start getting bunch of deprecation messages and your first thought is "Oh, FFS, why are you renaming stuff for no good reason". If there's an explanation, people might actually pay attention to their code when they do the renames and notice cases where they waste performance calculating txid over and over.

Reply:
Ok good point.  I've updated the dep messages for each renamed method.